numpyro/infer/mcmc.py

```diff
@@ -677,16 +755,26 @@ def run(self, rng_key, *args, extra_fields=(), collect_warmup=False, init_params
         assert isinstance(extra_fields, (tuple, list))
         collect_fields = tuple(set(('z', 'diverging') + tuple(extra_fields)))
         if self.num_chains == 1:
-            states_flat = self._single_chain_mcmc((rng_key, init_params), collect_fields, collect_warmup,
-                                                  args, kwargs)
+            states_flat = self._single_chain_mcmc(rng_key, init_params, args, kwargs, collect_fields, collect_warmup)
             states = tree_map(lambda x: x[np.newaxis, ...], states_flat)
         else:
             rng_keys = random.split(rng_key, self.num_chains)
-            partial_map_fn = partial(self._single_chain_mcmc,
-                                     collect_fields=collect_fields,
-                                     collect_warmup=collect_warmup,
-                                     args=args,
-                                     kwargs=kwargs)
+            if self._jit_model_args:
+                args = tree_map(lambda x: np.tile(x, (self.num_chains, 1)), args)
+                kwargs = tree_map(lambda x: np.tile(x, (self.num_chains, 1)), kwargs)
+                partial_map_fn = partial(self._single_chain_jit_args,
+                                         collect_fields=collect_fields,
+                                         collect_warmup=collect_warmup)
+            else:
+                # XXX: This is unfortunately still needed. Is there a better pattern?
```

Comment:
Thanks for the suggestion! I think with this, we probably don't even need to care about this broadcasting stuff since these parameters remain the same for all the chains.